app/src/main/java/com/example/nidoqueue/controller/RequestManager.java

```diff
@@ -1,64 +1,95 @@
 package com.example.nidoqueue.controller;
 
+import android.annotation.SuppressLint;
+import android.app.Application;
 import android.content.Intent;
 import android.provider.Settings;
 
 import com.example.nidoqueue.DataCalc;
-import com.example.nidoqueue.model.Database;
-import com.example.nidoqueue.model.Experiment;
 import com.example.nidoqueue.R;
-import com.example.nidoqueue.model.User;
 import com.example.nidoqueue.activity.AbstractActivity;
 import com.example.nidoqueue.activity.MainActivity;
 import com.example.nidoqueue.activity.SearchActivity;
 import com.example.nidoqueue.activity.SignInActivity;
 import com.example.nidoqueue.activity.UserProfileActivity;
+import com.example.nidoqueue.model.Experiment;
+import com.example.nidoqueue.model.User;
+import com.google.firebase.firestore.FirebaseFirestore;
 
-import static com.example.nidoqueue.controller.UserControl.contextManager;
-
-public class RequestManager {
+public class RequestManager extends Application {
 
     // Singleton pattern
-    private static final RequestManager requestManager = new RequestManager();
-    private RequestManager(){}
+    private static RequestManager requestManager;
+    private ContextManager contextManager;
+    private DatabaseManager databaseManager;
+    private UserControl userControl;
+    private ExperimentManager experimentManager;
+
+    @Override
+    public void onCreate() {
```

Comment:
I don't see why we need to make the Control layer into activities.  If we want to decrease coupling then we could make an abstract Controller class that all of the control classes extend.

Reply:
I just showed other way to implement singleton like stuff in Android, storing shared attributes
We can keep singleton. anyway this is not a "Activity".